src/libsyntax/print/pprust.rs

```diff
@@ -2334,22 +2337,24 @@ impl<'a> State<'a> {
     fn print_path(&mut self,
                   path: &ast::Path,
                   colons_before_params: bool,
-                  depth: usize)
+                  depth: usize,
+                  defaults_to_global: bool)
                   -> io::Result<()>
     {
         try!(self.maybe_print_comment(path.span.lo));
 
-        let mut first = !path.global;
-        for segment in &path.segments[..path.segments.len()-depth] {
-            if first {
-                first = false
-            } else {
+        let mut segments = path.segments[..path.segments.len()-depth].iter();
+        if defaults_to_global && path.segments[0].identifier.name == keywords::CrateRoot.name() {
```

Comment:
I think I would like an `is_global` method on paths, rather than having checks like this (`path.segments[0].identifier.name == keywords::CrateRoot.name()`) scattered around the code. It will also make it easier for tools using the AST.